source/features/fork-source-link-same-view.tsx

```diff
@@ -7,15 +7,30 @@ import doesFileExist from '../github-helpers/does-file-exist';
 import getDefaultBranch from '../github-helpers/get-default-branch';
 import {getRepo, getForkedRepo} from '../github-helpers';
 
+const isFilePath = (): boolean => pageDetect.isSingleFile()
+	|| (pageDetect.isRepoTree() && !pageDetect.isRepoRoot())
+	|| pageDetect.hasFileEditor();
+
 async function init(): Promise<void> {
 	const forkedRepository = getRepo(getForkedRepo())!;
 	const sameViewUrl = new GitHubURL(location.href).assign({
 		user: forkedRepository.owner,
 		repository: forkedRepository.name,
-		branch: await getDefaultBranch(forkedRepository),
 	});
 
-	if (await doesFileExist(sameViewUrl)) {
+	if (isFilePath()) {
+		sameViewUrl.branch = await getDefaultBranch(forkedRepository);
+	}
+
+	if (pageDetect.isIssue() || pageDetect.isPR()) {
+		sameViewUrl.assign({
+			route: '',
+			branch: '',
+			filePath: '',
+		});
+	}
+
+	if (!isFilePath() || await doesFileExist(sameViewUrl)) {
```

Comment:
Doesn't this trigger an extra request for every fork page we visit? I'm surprised we went with this solution.

Reply:
We did since it's a tiny request and we did not want 404s